vm/src/eval.rs

```diff
@@ -277,17 +328,37 @@ pub fn eval_inst(vm: &mut NexusVM<impl Memory>) -> Result<()> {
         ECALL { rd } => {
             RD = rd;
             vm.Z = vm.syscalls.syscall(vm.regs.pc, vm.regs.x, &vm.mem)?;
+            // Profile cycles
+            if vm.Z == SyscallCode::ProfileCycles as u32 {
+                handle_profile_cycles(vm)?;
+            }
```

Comment:
Reading `regs.x[18]` before the `syscall()` is the more future-proof way to tell the syscallcode.

Reply:
Good idea. Updated in https://github.com/nexus-xyz/nexus-zkvm/pull/243/commits/25013349658e9b04816cee6f148b723cd4bae97b